### PR TITLE
DBZ-3256 Consolidate Oracle redo/archive log queries

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-3256

This PR addresses moving the log resolution to a single query to address the following concerns:

* Moves de-duplication to database using SQL query, i.e. no duplicate SCN ranges
* Provides a consistent view of logs rather than relying on multiple queries

### CODE REVIEW

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/SqlUtils.java [hunk 1]
```diff
@@ -145,35 +145,38 @@ static String oldestFirstChangeQuery(Duration archiveLogRetention) {
...
+    public static String allMinableLogsQuery(Scn scn, Duration archiveLogRetention) {
         final StringBuilder sb = new StringBuilder();
-        sb.append("SELECT NAME AS FILE_NAME, NEXT_CHANGE# AS NEXT_CHANGE, FIRST_CHANGE# AS FIRST_CHANGE ");
-        sb.append("FROM ").append(ARCHIVED_LOG_VIEW).append(" ");
-        sb.append("WHERE NAME IS NOT NULL ");
-        sb.append("AND ARCHIVED = 'YES' ");
-        sb.append("AND STATUS = 'A' ");
-        sb.append("AND NEXT_CHANGE# > ").append(scn).append(" ");
-        sb.append("AND DEST_ID IN (").append(localArchiveLogDestinationsOnlyQuery()).append(") ");
+        sb.append("SELECT MIN(F.MEMBER) AS FILE_NAME, L.FIRST_CHANGE# FIRST_CHANGE, L.NEXT_CHANGE# NEXT_CHANGE, L.ARCHIVED, ");
```
Comment (jpechane) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/SqlUtils.java:157]: I think it is time to start documenting the queries. COuld you please add a comment with the final query and describe the fragments explaining what each does? Thnaks

Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/SqlUtils.java:201]: Pushed a commit with the requested changes.

**Comment by gunnarmorling** [comment]
@Naros, anything to add here? Or is it g2g from your PoV?

**Review by gunnarmorling — APPROVED** [review]
LGTM, merging. Thx, @Naros!

**Pull request merged**